kpet/run.py

```diff
@@ -191,7 +191,7 @@ def generate(self, description, tree_name, arch_name,
 
         if lint:
             parser = etree.XMLParser(remove_blank_text=True, encoding="utf-8")
-            tree = etree.XML(text, parser)
+            tree = etree.XML(text.encode("utf-8"), parser)
```

Comment:
I think we expect the parsed XML specify the encoding now (in XML declaration), so maybe we should remove the encoding="utf-8" argument, and then maybe we won't need to encode the text anymore. Could you try that?

Reply:
I can remove the XMLParser() and tostring() argument but because the text will be unicode and XML() can't read the declaration in unicode and then switch to utf-8 I don't think I can remove the str.encode() part.

> ValueError: Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.

And since we want the declaration the other option is to use bytes (encode)

Reply:
Thank you, Ondrej. I'm having trouble thinking through this, but I'll take a closer look at it myself. Maybe there's a way :)

Reply:
Alright then, let me know if you figure something out :)
Thanks